servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -165,7 +165,7 @@ private static void configure(ReadOnlyTcpServerConfig config, boolean autoRead,
         bs.option(ChannelOption.SO_BACKLOG, config.backlog());
 
         // Set the correct ByteBufAllocator based on our BufferAllocator to minimize memory copies.
-        ByteBufAllocator byteBufAllocator = BufferUtils.getByteBufAllocator(bufferAllocator);
+        ByteBufAllocator byteBufAllocator = POOLED_ALLOCATOR;
         bs.option(ChannelOption.ALLOCATOR, byteBufAllocator);
```

Comment:
We currently have a check in `NettyChannelPublisher`:
https://github.com/apple/servicetalk/blob/3a67c9ff8e4a1999e45d0e0fa8458d958671f9f0/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyChannelPublisher.java#L79-L89

The only potential risk I see is that we wrap `ByteBuf` with `Buffer` interface or create `HttpHeaders` object with multiple `ByteBuf` slices inside.
1. This wrapping hides `ReferenceCounted` interface => check in `NettyChannelPublisher` won't see it.
2. Presence of `ReferenceCounted` interface does not mean that the `ByteBuf` is pooled. Our unpooled buffers also implement `ReferenceCounted`, but override `release()` method to do nothing.

We can enhance `HttpObjectDecoder` to check each incoming `ByteBuf` in this way:
```java
if (buffer.alloc().isDirectBufferPooled()) {
    throw new IllegalStateException("Unpooled ByteBuf was expected, but received the pooled one");
}
```
But not sure this is required because we always have `CopyByteBufHandler` before `HttpObjectDecoder`. I think we can keep it as is and rely on the check in `NettyChannelPublisher` and netty's leak detector.

If you think we need something else, where do you propose to add it? Because we behave differently for h2 and h1 and we wrap with our ST types after h1/h2 decoders, a simple handler after protocol decoders won't catch it.

Reply:
I suggested change for the **aceept** pipeline, I think you have described behavior for the child pipeline.